key/src/main/java/software/amazon/kms/key/BaseHandlerStd.java

```diff
@@ -156,4 +160,34 @@ protected void resourceStateCheck(final KeyMetadata keyMetadata) {
           return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.NotFound);
       return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.GeneralServiceException);
     }
+
+    /**
+     * A helper method for validating that the requested resource model transition is possible
+     */
+    protected static ProgressEvent<ResourceModel, CallbackContext> validateResourceModel(
+            final ProgressEvent<ResourceModel, CallbackContext> progress, final ResourceModel previousModel,
+            final ResourceModel model) {
+        // If the key is asymmetric, we cannot enable key rotation
+        if (!Objects.equals(model.getKeySpec(), CustomerMasterKeySpec.SYMMETRIC_DEFAULT.toString()) &&
+                model.getEnableKeyRotation())
+            throw new CfnInvalidRequestException("You cannot set the EnableKeyRotation property to true on asymmetric keys.");
+
+        // Update specific validation
+        if (previousModel != null) {
+            // If the key is disabled, and will continue to be, we cannot modify the EnableKeyRotation property
+            if (!previousModel.getEnabled() && !model.getEnabled() &&
+                    previousModel.getEnableKeyRotation() != model.getEnableKeyRotation()) {
+                throw new CfnInvalidRequestException("You cannot modify the EnableKeyRotation property when the Enabled property is false. Set Enabled to true to modify the EnableKeyRotation property.");
```

Comment:
```suggestion
                throw new CfnInvalidRequestException("You cannot change the EnableKeyRotation property while the Enabled property is false.");
```

I don't think the second sentence is necessary.